server/plugin.go

```diff
@@ -65,7 +65,8 @@ type Plugin struct {
 
 	// A map of callID -> *cluster.Mutex to guarantee atomicity of call state
 	// operations.
-	callsClusterLocks map[string]*cluster.Mutex
+	callsClusterLocks    map[string]*cluster.Mutex
+	callsClusterLocksMut sync.RWMutex
```

Comment:
We're using a mut for a map of mutexes... 😂

Reply:
indeed, better than deadlocking which was the alternative :)